misaligned/misaligned.java

```diff
@@ -1,18 +1,180 @@
+import java.util.LinkedList;
+import java.util.List;
+
+import com.tcq.training.colourCode.ColourCode;
+
 public class Misaligned {
-    static int printColorMap() {
-        String majorColors[] = {"White", "Red", "Black", "Yellow", "Violet"};
-        String minorColors[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-        int i = 0, j = 0;
-        for(i = 0; i < 5; i++) {
-            for(j = 0; j < 5; j++) {
-                System.out.printf("%d | %s | %s\n", i * 5 + j, majorColors[i], minorColors[i]);
-            }
-        }
-        return i * j;
-    }
-    public static void main(String[] args) { 
-        int result = printColorMap();
-        assert(result == 25);
-        System.out.println("All is well (maybe!)");
-    }
+
+	void printColorMap(List<ColourCode> listOfColourCode) {
+		for (ColourCode colourCode : listOfColourCode) {
+			System.out.printf("%d  | %s | %s\n", colourCode.getPairNumber(), colourCode.getMajorColor(),
+					colourCode.getMinorColor());
+		}
+
+	}
+
+	List<ColourCode> generateColourCode() {
+		String majorColors[] = { "White", "Red", "Black", "Yellow", "Violet" };
+		String minorColors[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+		List<ColourCode> listOfColourCode = new LinkedList<ColourCode>();
+		int i = 0, j = 0;
+		for (i = 0; i < 5; i++) {
+			for (j = 0; j < 5; j++) {
+				listOfColourCode.add(new ColourCode.Builder().setPairNumber(i * 5 + j).setMajorColor(majorColors[i])
+						.setMinorColor(minorColors[i]).build());
+			}
+		}
+		return listOfColourCode;
+	}
+
+	public void testColourCode(List<ColourCode> listOfColourCode) {
+		int i = 1;
+		for (ColourCode colourCode : listOfColourCode) {
+
+			assertColourCode(i, colourCode);
+			i++;
+		}
+
+	}
+
+	public void assertColourCode(int key, ColourCode colourCode) {
+		switch (key) {
+		case 1:

```

Comment:
good approach to verify of the content of color map 👍 
An improvement can be that the test code may have to the case when number of major colors and number or minor colors change.